Client/Frontend/Widgets/ThumbnailCell.swift

```diff
@@ -36,47 +46,101 @@ class ThumbnailCell: UICollectionViewCell {
         return UILongPressGestureRecognizer(target: self, action: "SELdidLongPress")
     }()
 
-    let textLabel = UILabel()
-    let imageView = UIImageViewAligned()
-    let imageWrapper = UIView()
-    let removeButton = UIButton()
+    lazy var textWrapper: UIView = {
+        let wrapper = UIView()
+        wrapper.backgroundColor = ThumbnailCellUX.LabelBackgroundColor
+        return wrapper
+    }()
 
-    override init(frame: CGRect) {
-        imagePadding = CGFloat(0)
-        super.init(frame: frame)
+    lazy var textLabel: UILabel = {
+        let textLabel = UILabel()
+        textLabel.setContentHuggingPriority(1000, forAxis: UILayoutConstraintAxis.Vertical)
+        textLabel.font = ThumbnailCellUX.LabelFont
+        textLabel.textColor = ThumbnailCellUX.LabelColor
+        textLabel.textAlignment = ThumbnailCellUX.LabelAlignment
+        return textLabel
+    }()
 
-        addGestureRecognizer(longPressGesture)
+    lazy var imageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.contentMode = UIViewContentMode.ScaleAspectFit
 
-        contentView.addSubview(textLabel)
-        contentView.addSubview(imageWrapper)
-        imageWrapper.addSubview(imageView)
-        contentView.addSubview(removeButton)
+        imageView.clipsToBounds = true
+        imageView.layer.cornerRadius = ThumbnailCellUX.CornerRadius
+        return imageView
+    }()
 
+    lazy var backgroundImage: UIImageView = {
+        let backgroundImage = UIImageView()
+        backgroundImage.contentMode = UIViewContentMode.ScaleAspectFill
+        return backgroundImage
+    }()
+
+    lazy var backgroundEffect: UIVisualEffectView = {
+        let blur = UIBlurEffect(style: UIBlurEffectStyle.Light)
+        let vib = UIVibrancyEffect(forBlurEffect: blur)
+        return UIVisualEffectView(effect: blur)
+    }()
+
+    lazy var imageWrapper: UIView = {
+        let imageWrapper = UIView()
         imageWrapper.layer.borderColor = ThumbnailCellUX.BorderColor.CGColor
-        imageWrapper.layer.borderWidth = 1
-        imageWrapper.layer.cornerRadius = 3
+        imageWrapper.layer.borderWidth = ThumbnailCellUX.BorderWidth
+        imageWrapper.layer.cornerRadius = ThumbnailCellUX.CornerRadius
         imageWrapper.clipsToBounds = true
+        return imageWrapper
+    } ()
 
+    lazy var removeButton: UIButton = {
+        let removeButton = UIButton()
         removeButton.setImage(UIImage(named: "TileCloseButton"), forState: UIControlState.Normal)
         removeButton.addTarget(self, action: "SELdidRemove", forControlEvents: UIControlEvents.TouchUpInside)
         removeButton.hidden = true
         removeButton.imageEdgeInsets = ThumbnailCellUX.RemoveButtonInsets
+        return removeButton
+    } ()
+
+    override init(frame: CGRect) {
+        imagePadding = CGFloat(0)
```

Comment:
Since we're setting up the constraints below, we probably don't want the didSet handler to get called whenever we set this to 0 here. Instead we set it to 0 in the declaration. Also, I don't see it in this diff but inside the didSet handler, after we call snp_remakeConstraints on the imageView, we should setNeedsUpdateConstraints on the imageWrapper to notify it of the constraint change on it's next layout pass.